File: exam-slammr-webapp/src/main/java/uk/co/claritysoftware/exam/slammr/webapp/web/model/user/SocialUserSignUp.java
package uk.co.claritysoftware.exam.slammr.webapp.web.model.user;

import javax.validation.constraints.Email;
import javax.validation.constraints.NotBlank;
import org.hibernate.validator.constraints.Length;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Class encapsulating fields to support a user sign up from a social provider
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public final class SocialUserSignUp {

	@NotBlank
	@Length(max = 30)
	private String firstName;

	@NotBlank
	@Length(max = 30)
	private String lastName;

	@NotBlank
	@Length(max = 30)
	private String nickName;

	@NotBlank
	@Email
	@Length(max = 64)
	private String email;
}

File: exam-slammr-webapp/src/main/kotlin/uk/co/claritysoftware/exam/slammr/webapp/web/model/user/SocialUserSignUp.kt
package uk.co.claritysoftware.exam.slammr.webapp.web.model.user

import org.hibernate.validator.constraints.Length
import javax.validation.constraints.Email
import javax.validation.constraints.NotBlank

/**
 * Class encapsulating fields to support a user sign up from a social provider
 */
data class SocialUserSignUp(
		@field:NotBlank
		@field:Length(max = 30)
		var firstName: String,

		@field:NotBlank
		@field:Length(max = 30)
		var lastName: String,

		@field:NotBlank
		@field:Length(max = 30)
		var nickName: String,

		@field:NotBlank
		@field:Email
		@field:Length(max = 64)
		var email: String
)
